ui/src/app/components/app-router.tsx

```diff
@@ -16,8 +16,18 @@ export const usePath = () => {
   return path;
 };
 
-export const withRouteData = WrappedComponent => ({routeData, ...props}) => {
-  routeDataStore.set(routeData);
+// TODO angular2react: I don't think this is really the right place to be making the store updates but it's the
+// best place I found while we're using both angular and react routers
+export const withRouteData = WrappedComponent => ({intermediaryRoute = false, routeData, ...props}) => {
+  const params = useParams();
+  console.log('using params');
+
+  if (!intermediaryRoute) {
```

Comment:
This should probably be inside a `useEffect()`, which reruns based on `params` changing. This render method's main body would otherwise run an arbitrary number of times, at non-deterministic timings.

In the future, I think urlParamsStore should be removed entirely in favor of `useParams`. For the route data, I'm not sure of the best pattern, though we should likely switch off this global state. Anyways - this seems good during the transition.